### PR TITLE
First commit of updated torch code

### PR DESCRIPTION
**Pull request by jsschreck**
Torch training and predict scripts are now available in applications, along with a new config file. 

The workflow here mimics our AI-NWP workflow, which enables scaling. The model is still a simple MLP. The user needs to supply a torch dataset, which mlguess will import from a custom import. In the future we can extend this to models and trainer classes if needed.

### CODE REVIEW

**Comment by djgagne** [comment]
@jsschreck Can you add unit tests and docstrings, especially for the model, layer, and loss classes? The docstrings should be in google format and should match formatting on the keras side of things. There are already unit tests in the mlguess/tests directory for you to emulate.

**Comment by djgagne** [comment]
I made a bunch of formatting updates to the code, got the docs to compile, removed the ptype-physical dependency and added some other ones, and removed an accidental credit dependency. There are still docstring and formatting fixes that need to be done, but the new torch code seems to work and to play nicely with keras on a torch backend.

**Review by djgagne — APPROVED** [review]
After my edits, CI is now passing. Given the size of this PR, I am going to go ahead and approve and then work on further cleanup in a future PR.
